src/components/web-form-renderer.vue

```diff
@@ -0,0 +1,336 @@
+    headers: {
+      'content-type': 'text/xml'
+    },
+    fulfillProblem: () => true
+  })
+    .then(({ data }) => {
+      if (isProblem(data)) {
+        primaryInstance.uploadSuccess = false;
+        submissionModal.errorMessage = data.message;
+        showModal('error');
+        return false;
+      }
+      Object.assign(primaryInstance, { instanceId: data.instanceId, uploadSuccess: true });
+      return true;
+    })
+    .catch(noop);
+};
+
+const uploadAttachment = async (attachmentIndex) => {
+  const attachmentDescriptor = attachmentUploads[attachmentIndex];
+  let url = apiPaths.submissionAttachment(form.projectId, form.xmlFormId, false, primaryInstance.instanceId, attachmentDescriptor.file.name);
+  if (form.draftToken) {
+    url = url.replace(/^\/v1\//, `/v1/test/${form.draftToken}/`);
+  }
+  if (route.query.st) {
+    url += `?st=${route.query.st}`;
+  }
+  await request({
+    method: 'POST',
+    url,
+    data: attachmentDescriptor.file,
+    onUploadProgress: (progressEvent) => {
+      attachmentDescriptor.progress = Math.floor((progressEvent.loaded * 100) / progressEvent.total);
+    },
+  })
+    .then(() => {
+      attachmentDescriptor.uploadSuccess = true;
+    })
+    .catch(() => {
+      Object.assign(attachmentDescriptor, { uploadSuccess: false, progress: null });
+    });
+};
+
+const postSubmission = async () => {
+  // TODO: add logic for edit submission
+
+  if (await postPrimaryInstance()) {
+    // eslint-disable-next-line no-unused-vars
+    for (const [ix, _] of attachmentUploads.entries()) {
+      // Post each attachment in turn, and not in parallel. Parallel transfers means more time spent per transfer, which
+      // under adverse network conditions increases the probability that the transfer does not complete. And since uploads
+      // as of yet are not resumable, this would mean restarting the upload from 0 again.
+      // eslint-disable-next-line no-await-in-loop
+      await uploadAttachment(ix);
+    }
+
+    return true;
+  }
+
+  return false;
+};
+
+
+/**
+ * When WebForms's submit button is clicked, it dispatches an event which is handed to
+ * this handler, which can then upload the form and its attachments as present in the
+ * event payload.
+ */
+const handleSubmit = async (payload) => {
+  // TODO: waiting for web-forms v0.7, current version doesn't return any payload.
+  // eslint-disable-next-line no-constant-condition
+  if (props.actionType === 'preview') {
+    showModal('preview');
+  } else {
+    // eslint-disable-next-line no-unused-vars
+    const { data, definition, status, violations } = payload;
+    if (status !== 'ready') {
+      // Status is not ready when Form is not valid and in that case submit button will be disabled,
+      // hence this branch should never execute.
+      return;
+    }
+
+    primaryInstance.file = data.instanceFile;
+    data.attachments.forEach((file, ix) => { attachmentUploads[ix] = { file, progress: null, uploadSuccess: null }; });
```

Comment:
Don't we need to resize the array at some point? Alternatively, maybe we could use `attachmentUploads.push()` here. It seems to me that it's possible that `attachmentUploads` is an empty array here, in which case I don't think we should directly set an index on it like what's being done here.